virttest/test_setup.py

```diff
@@ -434,14 +434,19 @@ def __init__(self, params):
         vm_mem_minimum = params.get("vm_mem_minimum", "512M")
         self.lowest_mem_per_vm = float(normalize_data_size(vm_mem_minimum))
         utils_memory.drop_caches()
+        self.hugepages_surp = int(params.get("hugepages_surp", 0))
+        over_commit_hp = "/proc/sys/vm/nr_overcommit_hugepages"
+        over_commit = kernel_interface.ProcFS(over_commit_hp)
+        if self.hugepages_surp != over_commit.proc_fs_value:
+            over_commit.proc_fs_value = self.hugepages_surp
 
         target_hugepages = params.get("target_hugepages")
         if target_hugepages is None:
             target_hugepages = self.get_target_hugepages()
         else:
             target_hugepages = int(target_hugepages)
 
-        self.target_hugepages = target_hugepages
+        self.target_hugepages = target_hugepages - self.hugepages_surp
```

Comment:
Is it possible the surplus hugepage is being used by other process,  then it should be `        self.target_hugepages = target_hugepages + self.hugepages_surp` ?

Reply:
> Is it possible the surplus hugepage is being used by other process

Yes, you're right. And @balamuruhans means requesting a new parameter to test surplus hugepage.
So maybe we need 1 new parameter and 2 new variables of it.
In cfg:
```
surp_hugepage = int
```
In py:
```python
self.ext_surp_hugepage = int(utils_memory.get_num_huge_pages_surp())
hugepages_surp = int(params.get("hugepages_surp", 0))
self.target_hugepages = target_hugepages + self.ext_surp_hugepage - hugepages_surp
```